mkosi/kmod.py

```diff
@@ -92,6 +123,19 @@ def normalize_module_name(name: str) -> str:
     return name.replace("_", "-")
 
 
+def normalize_module_glob(name: str) -> str:
+    # We want to replace '_' by '-', except when used in […]
```

Comment:
That doesn't treat `_` and `-` equivalently, it only treats all `_` as `-`, but not the other way around? `intel-vsec` is a module where both `modinfo intel-vsec` and `modinfo intel_vsec` work, but the name is `intel_vsec`. I'm not sure it's a good idea to replace the correct name by something that also works, this seems like something that's bound to break. Maybe this should replace both `-` and `_` by `[-_]`?

Reply:
The "normalization" only needs to be done in one direction, as long as it's the same direction. We call `normalize_module_glob()` on the globs, and `normalize_module_name()` on module names. They both change `_` to `-`, and then we get a match. This change is not visible to the user, except that patterns with both characters match module names. This is what the kernel implements too.

The normalization happens in filtering, but the actual path that is used is the path found on the file system, i.e. the actual module name.